### DISCUSSION TITLE
why Twin does not use TwinState?

### DISCUSSION THREAD

**Discussion by ranjith-t** [Azure/azure-iot-sdk-java#1609] [Ideas]
In azure-iot-sdk-javaj version 2, iot-service-client module, I could see that the Twin class handles the device twin. While this has same information repeated in TwinState. Is there a plan ot use TwinState in Twin?
Also, there is no API for toJSON*() in Twin? Is there a plan to add one in future?

**Comment by timtay-microsoft** [comment]
You are correct that there are some redundancies between Twin and TwinState even in v2. We don't currently have any plans for a v3 to make breaking changes to address this, though.

We don't have any plans to add a twin.toJson() method off of the Twin class, no. Is there some reason you would need it?

**Reply by ranjith-t** [reply]
ok, thanks for your reply. 
Regarding toJSON(): yes we would need a json converted data of the twin to be stored locally(and use that to initialize the Twin object later when needed). There is a fromJson() API available, but that is not useful unless we have a toJSON()? The toString() does not help as it just converts it to String and not JSON. 

As a work around I am using gson() to convert via "gson.toJson(newtwin)", but afraid that is a generic API and we may miss information.

**Reply by timtay-microsoft** [reply]
Your workaround seems safe to me. If we were to expose a ```twin.toJson()``` method, that would be the implementation.
